packages/contracts/contracts/program/ProgramFactory.sol

```diff
@@ -34,27 +34,28 @@ contract ProgramFactory is OwnableUpgradeable {
    * contract while relying on the same ProgramFactory to get the list of
    * programs.
    */
-  function updateProgramContract(address _programContract) public onlyOwner {
-    programContract = _programContract;
+  function updateProgramContract(address newProgramContract) external onlyOwner {
+    // slither-disable-next-line missing-zero-check
+    programContract = newProgramContract;
 
-    emit ProgramContractUpdated(_programContract);
+    emit ProgramContractUpdated(newProgramContract);
   }
 
   /**
    * @notice Clones ProgramImplmentation and deployed a program and emits an event
    *
-   * @param _encodedParameters Encoded parameters for creating a program
+   * @param encodedParameters Encoded parameters for creating a program
    */
   function create(
-    bytes calldata _encodedParameters
+    bytes calldata encodedParameters
   ) external returns (address) {
 
     address clone = ClonesUpgradeable.clone(programContract);
 
-    ProgramImplementation(clone).initialize(_encodedParameters);
-
     emit ProgramCreated(clone);
 
+    ProgramImplementation(clone).initialize(encodedParameters);
```

Comment:
wouldn't we want to emit after the initialize has happened ?

Reply:
from an EVM perspective, it doesn't matter where events are emitted, however, in the case, it was re-entrant (unlikely), the event would be emitted the correct number of times.